flow_runner.go

```diff
@@ -142,37 +148,121 @@ func (f *flowRunner) runTask(ctx context.Context, task Task) bool {
 		return true
 	}
 
-	w := f.output
-	if !f.verbose {
-		w = &strings.Builder{}
+	paramValues := make(map[string]Value)
+	for _, param := range task.Parameters {
+		paramValues[param.Name()] = f.paramValues[param.Name()]
+	}
+	if f.verbose != nil {
+		paramValues[f.verbose.Name()] = f.paramValues[f.verbose.Name()]
 	}
 
-	// report task start
-	fmt.Fprintf(w, "===== TASK  %s\n", task.Name)
+	failed := false
+	measuredCommand := func(tf *TF) {
+		w := tf.Output()
+		if (f.verbose != nil) && !f.verbose.Get(tf) {
+			w = &strings.Builder{}
+		}
 
-	// run task
-	runner := Runner{
-		Ctx:      ctx,
-		TaskName: task.Name,
-		Verbose:  f.verbose,
-		Params:   f.params,
-		Output:   w,
+		// report task start
+		fmt.Fprintf(w, "===== TASK  %s\n", tf.Name())
+
+		// run task
+		runner := Runner{
+			Ctx:         tf.Context(),
+			TaskName:    tf.Name(),
+			ParamValues: tf.paramValues,
+			Output:      w,
+		}
+		result := runner.Run(task.Command)
+
+		// report task end
+		status := "PASS"
+		switch {
+		case result.Failed():
+			status = "FAIL"
+			failed = true
+		case result.Skipped():
+			status = "SKIP"
+		}
+		fmt.Fprintf(w, "----- %s: %s (%.2fs)\n", status, tf.Name(), result.Duration().Seconds())
+
+		if sb, ok := w.(*strings.Builder); ok && result.failed {
+			io.Copy(tf.Output(), strings.NewReader(sb.String())) //nolint // not checking errors when writting to output
+		}
+	}
+
+	measuredRunner := Runner{
+		Ctx:         ctx,
+		TaskName:    task.Name,
+		ParamValues: paramValues,
+		Output:      f.output,
+	}
+	measuredRunner.Run(measuredCommand)
+
+	return !failed
+}
+
+func (f *flowRunner) unusedParams() []string {
+	remainingParams := make(map[string]struct{})
+	for key := range f.params {
+		remainingParams[key] = struct{}{}
 	}
-	result := runner.Run(task.Command)
+	if f.verbose != nil {
+		delete(remainingParams, f.verbose.Name())
+	}
+	for _, task := range f.tasks {
+		for _, param := range task.Parameters {
+			delete(remainingParams, param.Name())
+		}
+	}
+	unusedParams := make([]string, 0, len(remainingParams))
+	for key := range remainingParams {
+		unusedParams = append(unusedParams, key)
+	}
+	return unusedParams
+}
 
-	// report task end
-	status := "PASS"
-	switch {
-	case result.Failed():
-		status = "FAIL"
-	case result.Skipped():
-		status = "SKIP"
+func printUsage(f *flowRunner) {
+	fmt.Fprintf(f.output, "Usage: [flag(s)] task(s)\n")
```

Comment:
Thanks to your implementation the flags do not have to be before tasks.
How about changing this line to:
```suggestion
	fmt.Fprintf(f.output, "Usage: task(s) [flag(s)]\n")
```

The README would also need to be updated.

Reply:
I've updated it to highlight any sequence.